src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -818,15 +829,26 @@ def _add_new_text_input_to_json(
 
     @classmethod
     def _add_new_prompt_to_json(
-        cls, pa_json: Dict, index: int, new_prompt: str
+        cls,
+        pa_json: Dict,
+        index: int,
+        new_prompt: str,
+        concatenate_prompts: bool = True,
     ) -> Dict:
         if new_prompt:
             if pa_json["data"][index]["payload"][0]["prompt"][0]:
-                pa_json["data"][index]["payload"][0]["prompt"][0] = (
-                    pa_json["data"][index]["payload"][0]["prompt"][0] + f" {new_prompt}"
-                )
+                if concatenate_prompts:
+                    pa_json["data"][index]["payload"][0]["prompt"][0] = (
+                        pa_json["data"][index]["payload"][0]["prompt"][0]
+                        + f" {new_prompt}"
+                    )
+                else:
+                    pa_json["data"][index]["payload"][0]["prompt"][0].append(new_prompt)
             else:
-                pa_json["data"][index]["payload"][0]["prompt"][0] = new_prompt
+                if concatenate_prompts:
```

Comment:
Can you make sure to test that this works? We might also need a comment here explaining why this branching is needed. My thought would be that this would just break due to the incorrect format, though I assume this has to do with adding string support then using it for some reason in this case.